enf-reducer/rte-helpers.metta

```diff
@@ -137,16 +137,10 @@
 
 ;;This function applies a function to each values of the first list and the second list.
 ;;The second list will be repeatedly passed without changing while the elements of the first list change.
-(: mapList (-> Atom (List $t) (List $t) (List $t)))
-(= (mapList $f Nil $rs) Nil)
-(= (mapList $f (Cons $x $xs) $rs)
-  (Cons ($f $x $rs) (mapList $f $xs $rs))
-)
-;;A function that applies a function to the elements of the first list and a constant parent.
-(: mapParent (-> $f (List $t) $t (List $tt)))
-(= (mapParent $f Nil $parent) Nil)
-(= (mapParent $f (Cons $x $xs) $parent)
-  (Cons ($f $x $parent) (mapParent $f $xs $parent))
+(: binaryMap (-> Atom (List $t) (List $t) (List $t)))
```

Comment:
It's best if you avoid using `Atom` as type. You can define the type of the function that will be accepted as `(-> $t $t $t)`.

Reply:
And shouldn't the second argument for the `binaryMap` function should be of type `$t` instead of `(List $t)`?

Reply:
You can then use this `binaryMap` instead of mapParent :)